security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -237,12 +236,6 @@ func (s *sdsservice) StreamSecrets(stream sds.SecretDiscoveryService_StreamSecre
 				return receiveError
 			}
 
-			if discReq.Node == nil {
```

Comment:
This was added by https://github.com/istio/istio/pull/16394, and that PR mentioned that, only the first SDS request is guaranteed to have node identifier, followup SDS requests may have empty descReq.Node. If descReq.Node is null, we need to assign it to node. Otherwise, the following logic would break.

Reply:
only if `set_node_on_first_message_only=true` which it isn't